server/types/interval.go

```diff
@@ -0,0 +1,277 @@
+	return false
+}
+
+// MaxSerializedWidth implements the DoltgresType interface.
+func (b IntervalType) MaxSerializedWidth() types.ExtendedTypeSerializedWidth {
+	return types.ExtendedTypeSerializedWidth_64K
+}
+
+// MaxTextResponseByteLength implements the DoltgresType interface.
+func (b IntervalType) MaxTextResponseByteLength(ctx *sql.Context) uint32 {
+	return 64
+}
+
+// OID implements the DoltgresType interface.
+func (b IntervalType) OID() uint32 {
+	return uint32(oid.T_interval)
+}
+
+// Promote implements the DoltgresType interface.
+func (b IntervalType) Promote() sql.Type {
+	return Interval
+}
+
+// SerializedCompare implements the DoltgresType interface.
+func (b IntervalType) SerializedCompare(v1 []byte, v2 []byte) (int, error) {
+	if len(v1) == 0 && len(v2) == 0 {
+		return 0, nil
+	} else if len(v1) > 0 && len(v2) == 0 {
+		return 1, nil
+	} else if len(v1) == 0 && len(v2) > 0 {
+		return -1, nil
+	}
+
+	d1, err := deserializeDuration(v1)
+	if err != nil {
+		return 0, err
+	}
+	d2, err := deserializeDuration(v2)
+	if err != nil {
+		return 0, err
+	}
+	return d1.Compare(d2), nil
+}
+
+// SQL implements the DoltgresType interface.
+func (b IntervalType) SQL(ctx *sql.Context, dest []byte, v any) (sqltypes.Value, error) {
+	if v == nil {
+		return sqltypes.NULL, nil
+	}
+	value, err := b.IoOutput(ctx, v)
+	if err != nil {
+		return sqltypes.Value{}, err
+	}
+	return sqltypes.MakeTrusted(sqltypes.Text, types.AppendAndSliceBytes(dest, []byte(value))), nil
+}
+
+// String implements the DoltgresType interface.
+func (b IntervalType) String() string {
+	return "interval"
+}
+
+// ToArrayType implements the DoltgresType interface.
+func (b IntervalType) ToArrayType() DoltgresArrayType {
+	return IntervalArray
+}
+
+// Type implements the DoltgresType interface.
+func (b IntervalType) Type() query.Type {
+	return sqltypes.Text
+}
+
+// ValueType implements the DoltgresType interface.
+func (b IntervalType) ValueType() reflect.Type {
+	return reflect.TypeOf(duration.MakeDuration(0, 0, 0))
+}
+
+// Zero implements the DoltgresType interface.
+func (b IntervalType) Zero() any {
+	return duration.MakeDuration(0, 0, 0)
+}
+
+// SerializeType implements the DoltgresType interface.
+func (b IntervalType) SerializeType() ([]byte, error) {
+	return SerializationID_Interval.ToByteSlice(0), nil
+}
+
+// deserializeType implements the DoltgresType interface.
+func (b IntervalType) deserializeType(version uint16, metadata []byte) (DoltgresType, error) {
+	switch version {
+	case 0:
+		return Interval, nil
+	default:
+		return nil, fmt.Errorf("version %d is not yet supported for %s", version, b.String())
+	}
+}
+
+// SerializeValue implements the DoltgresType interface.
+func (b IntervalType) SerializeValue(val any) ([]byte, error) {
+	if val == nil {
+		return nil, nil
+	}
+	converted, _, err := b.Convert(val)
+	if err != nil {
+		return nil, err
+	}
+	str := converted.(duration.Duration).String()
+	writer := utils.NewWriter(uint64(len(str) + 1))
```

Comment:
Left some feedback on an earlier comment for a better way to handle this (and `SerializedCompare`).